pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -53,10 +58,32 @@ type framework struct {
 	permitPlugins         []PermitPlugin
 }
 
-const (
-	// Specifies the maximum timeout a permit plugin can return.
-	maxTimeout time.Duration = 15 * time.Minute
-)
+// extensionPoint encapsulates desired and applied set of plugins at a specific extension
+// point. This is used to simplify iterating over all extension points supported by the
+// framework.
+type extensionPoint struct {
+	// the set of plugins to be configured at this extension point.
+	plugins *config.PluginSet
+	// a pointer to the slice storing plugins implementations that will run at this
+	// extenstion point.
+	slicePtr interface{}
+}
+
+func (f *framework) getExtensionPoints(plugins *config.Plugins) []extensionPoint {
```

Comment:
It seems like we back to use the "separate data and logic" approach.

Reply:
> It seems like we back to use the "separate data and logic" approach.

yes, but there is a difference between the reason we are doing it here and what you suggested in your PR: you separated the data but made it local to the function, you didn't identify or argued that there is an opportunity to reuse the "data" in other places, and so I still stand by my opinion that that wasn't useful and it only bloated the function.

The situation here is different: 1) we added another function (listPlugins) that needed to iterate over the same set of extension points, 2) identified a third existing function (pluginsNeeded) that also needed to iterate the same extension points.

Is the difference clear?

Reply:
Sorry，sir ，this name “PodToleratesNodeTaint” seems little Confuse to me。。。